packages/core-v2/src/semantic-model/in-memory/in-memory-semantic-model.ts

```diff
@@ -0,0 +1,188 @@
+import {InMemoryEntityModel} from "../../entity-model/implementation";
+import {
+    isCreateClassOperation, isCreateGeneralizationOperation,
+    isCreateRelationshipOperation,
+    isDeleteEntityOperation,
+    isModifyClassOperation, isModifyGeneralizationOperation,
+    isModifyRelationOperation,
+    Operation,
+    OperationResult
+} from "../operations/operations";
+import {SemanticModelClass, SemanticModelGeneralization, SemanticModelRelationship} from "../concepts/concepts";
+
+function uuid() {
+    return Math.random().toString(36).substring(2) + Date.now().toString(36);
+}
+
+export class InMemorySemanticModel extends InMemoryEntityModel {
+    public executeOperation(operation: Operation): OperationResult {
+        // todo this is just a mock implementation
+
+        if (isCreateClassOperation(operation)) {
+            let id = operation.entity.id;
+
+            // Generate random id if not provided
+            if (id === undefined) {
+                id = uuid();
+            }
+
+            if (this.entities[id]) {
+                return {
+                    success: false,
```

Comment:
Ups, chtel jsem ten komentar z radky 52 mit tady:
Predpokladam, ze tohle by se stavat nemelo, kdyz se budou dobre generovat uuids. Kdyz se to ale stane, tak by se mela snova vyvolat ta modifyClassOperation akce, ktera uz snad dopadne?